src/util/TickUtils.ts

```diff
@@ -0,0 +1,78 @@
+import _ from 'lodash';
+
+import { TickFormatter } from '../cartesian/CartesianAxis';
+import { getAngledRectangleWidth } from './CartesianUtils';
+import { getStringSize } from './DOMUtils';
+import { getEveryNthWithCondition } from './getEveryNthWithCondition';
+import { Size, CartesianViewBox, CartesianTickItem } from './types';
+
+export function getAngledTickWidth(contentSize: Size, unitSize: Size, angle: number) {
+  const size = { width: contentSize.width + unitSize.width, height: contentSize.height + unitSize.height };
+
+  return getAngledRectangleWidth(size, angle);
+}
+
+export function getInitialStartAndEnd(viewBox: CartesianViewBox, sign: number, sizeKey: string) {
+  let start, end;
+
+  const { x, y, width, height } = viewBox;
+  if (sign === 1) {
+    start = sizeKey === 'width' ? x : y;
+    end = sizeKey === 'width' ? x + width : y + height;
+  } else {
+    start = sizeKey === 'width' ? x + width : y + height;
+    end = sizeKey === 'width' ? x : y;
+  }
+  return { end, start };
+}
+export function getSizeOfTick(
+  tickFormatter: TickFormatter,
+  entry: CartesianTickItem,
+  index: number,
+  sizeKey: 'width' | 'height',
+  fontSize: string | number,
+  letterSpacing: string | number,
+  unitSize: Size,
+  angle: number,
+) {
+  const content = _.isFunction(tickFormatter) ? tickFormatter(entry.value, index) : entry.value;
+  // Recharts only supports angles when sizeKey === 'width'
+  return sizeKey === 'width'
+    ? getAngledTickWidth(getStringSize(content, { fontSize, letterSpacing }), unitSize, angle)
+    : getStringSize(content, { fontSize, letterSpacing })[sizeKey];
+}
+
+export function doesTickFitInBetweenStartAndEnd(
```

Comment:
nit wording: doesTickFitInBetweenStartAndEnd what? does tick fit between start and end tick?

Reply:
Start and end are numbers, which are updated as we traverse the ticks either from the end of the axis towards the start (for the strategy preserveEnd), or from the start to the end.